packages/server/src/sdk/app/views/index.ts

```diff
@@ -299,15 +292,24 @@ export async function enrichSchema(
 
   const viewSchema = view.schema || {}
   const anyViewOrder = Object.values(viewSchema).some(ui => ui.order != null)
-  for (const key of Object.keys(tableSchema).filter(
-    k => tableSchema[k].visible !== false
-  )) {
+
+  const visibleSchemaFields = Object.keys(viewSchema).filter(key => {
+    if (helpers.views.isCalculationField(viewSchema[key])) {
+      return viewSchema[key].visible !== false
+    }
+    return key in tableSchema && tableSchema[key].visible !== false
```

Comment:
Is this not bypassing the view visibility?

Reply:
I broke some tests while working on this and the current state of the code passes the various visibility tests I broke. I haven't changed any visibility tests. In what way would this break view visibility?

Reply:
Interesting... Good we have tests for this. I were just worried that we do `const visibleFields = new Set([...visibleSchemaFields, ...visibleTableFields])`, including the table fields. It might be trimmed in some post-process then